page_iterator.go

```diff
@@ -107,31 +105,32 @@ func (pI *PageIterator) SetReqOptions(reqOptions []abstractions.RequestOption) {
 	pI.reqOptions = reqOptions
 }
 
-func (pI *PageIterator) hasNext() bool {
-	if pI.currentPage == nil || pI.currentPage.getNextLink() == nil {
-		return false
+func (pI *PageIterator) next() error {
```

Comment:
next should be public
next should return the page

Reply:
There's still a discussion going on about this and I don't want that discussion to block this PR. Will open another issue if necessary to address concerns.